ush/run_jedi_exe.py

```diff
@@ -47,67 +67,137 @@ def run_jedi_exe(yamlconfig):
-        'fv3jedi_fix_dir': os.path.join(workdir, 'Data', 'fv3files'),
-        'fv3jedi_fieldmetadata_dir': os.path.join(workdir, 'Data', 'fieldmetadata'),
-        'ANL_DIR': os.path.join(workdir, 'anl'),
-        'fv3jedi_staticb_dir': os.path.join(workdir, 'berror'),
-        'BIAS_IN_DIR': os.path.join(workdir, 'obs'),
-        'BIAS_OUT_DIR': os.path.join(workdir, 'bc'),
-        'CRTM_COEFF_DIR': os.path.join(workdir, 'crtm'),
-        'BIAS_PREFIX': f"{executable_subconfig['dump']}.t{gcyc}z.",
-        'BIAS_DATE': f"{gdate}",
-        'DIAG_DIR': os.path.join(workdir, 'diags'),
-        'OBS_DIR': os.path.join(workdir, 'obs'),
-        'OBS_PREFIX': f"{executable_subconfig['dump']}.t{cyc}z.",
-        'OBS_DATE': f"{cdate}",
-        'valid_time': f"{valid_time.strftime('%Y-%m-%dT%H:%M:%SZ')}",
-        'window_begin': f"{window_begin.strftime('%Y-%m-%dT%H:%M:%SZ')}",
-        'prev_valid_time': f"{prev_cycle.strftime('%Y-%m-%dT%H:%M:%SZ')}",
-        'atm_window_length': executable_subconfig['atm_window_length'],
-        'CASE': executable_subconfig['case'],
-        'CASE_ANL': executable_subconfig.get('case_anl', executable_subconfig['case']),
-        'CASE_ENKF': executable_subconfig.get('case_enkf', executable_subconfig['case']),
-        'DOHYBVAR': executable_subconfig.get('dohybvar', False),
-        'LEVS': str(executable_subconfig['levs']),
-        'forecast_steps': calc_fcst_steps(executable_subconfig.get('forecast_step', 'PT6H'),
-                                          executable_subconfig['atm_window_length']),
-        'BKG_TSTEP': executable_subconfig.get('forecast_step', 'PT6H'),
-        'INTERP_METHOD': executable_subconfig.get('interp_method', 'barycentric'),
-    }
-    template = executable_subconfig['yaml_template']
-    output_file = os.path.join(workdir, f"gdas_{app_mode}.yaml")
-    # set some environment variables
-    os.environ['PARMgfs'] = os.path.join(all_config_dict['GDASApp home'], 'parm')
-    # generate YAML for executable based on input config
-    logging.info(f'Using YAML template {template}')
-    ufsda.yamltools.genYAML(var_config, template=template, output=output_file)
-    logging.info(f'Wrote YAML file to {output_file}')
-    # use R2D2 to stage backgrounds, obs, bias correction files, etc.
-    ufsda.stage.gdas_single_cycle(var_config)
-    # link additional fix files needed (CRTM, fieldmetadata, etc.)
-    gdasfix = executable_subconfig['gdas_fix_root']
-    ufsda.stage.gdas_fix(gdasfix, workdir, var_config)
-    # link executable
-    baseexe = os.path.basename(executable_subconfig['executable'])
-    ufsda.disk_utils.symlink(executable_subconfig['executable'], os.path.join(workdir, baseexe))
-    # create output directories
-    ufsda.disk_utils.mkdir(os.path.join(workdir, 'diags'))
-    if app_mode in ['variational']:
-        ufsda.disk_utils.mkdir(os.path.join(workdir, 'anl'))
-        ufsda.disk_utils.mkdir(os.path.join(workdir, 'bc'))
+    pdy = prev_cycle.strftime("%Y%m%d")
```

Comment:
Normally `pdy` is the current cycle.   If we want `pdy` to hold the previous cycle PDY, we should rename this variable `gpdy` to be consistent with `gcyc` and `gdate`.

Reply:
@RussTreadon-NOAA changed `pdy` as suggested.

Reply:
Thank you, @guillaumevernieres.   `pdy` goes with `valid_time`.  Commit at [49f8337](https://github.com/NOAA-EMC/GDASApp/commit/49f83372b4d3ca824ce10ea1b68933ee1409ef11) looks good.